src/main/java/org/fundacionjala/coding/omar/CesarEncryption.java

```diff
@@ -0,0 +1,178 @@
+import java.util.stream.Collectors;
+
+/**
+ * @author Omar Limbert Huanca Sanchez - AT-[06].
+ * @version 1.0.
+ */
+public class CesarEncryption {
+
+    private static final char LAST_LETTER = 'Z';
+    private static final char FIRST_LETTER = 'A';
+    private static final int DIFFERENCE = 26;
+    private int indexVigenereKey;
+
+    /**
+     * Constructor to initialize index for Vigenere Encryption.
+     */
+    public CesarEncryption() {
+        this.indexVigenereKey = 0;
+    }
+
+    /**
+     * Method to encode Message with Cesar Encryption.
+     *
+     * @param message This is message to encode.
+     * @param key     This is key for encode.
+     * @return String , string already encode with Cesar Encryption.
+     */
+    public String encodeMessageWithCesarEncryption(String message, int key) {
+        return message == null || message.trim().isEmpty()
+                ? ""
+                : Arrays.stream(message.split(""))
+                .map(letter -> this.algorithm(letter, key))
+                .collect(Collectors.joining());
+    }
+
+    /**
+     * Method to decode Message with Cesar Encryption.
+     *
+     * @param message This is message to decode.
+     * @param key     This is key for decode.
+     * @return String , string already encode with Cesar Encryption.
+     */
+    public String decodeMessageWithCesarEncryption(String message, int key) {
+        return message == null || message.trim().isEmpty()
+                ? ""
+                : Arrays.stream(message.split(""))
+                .map(letter -> this.algorithm(letter, -key))
+                .collect(Collectors.joining());
+    }
+
+    /**
+     * Method to encode Message with Vigenere Encryption.
+     *
+     * @param message This is message to encode with Vigenere Encryption.
+     * @param key     This is a key with string value.
+     * @return String , string already encode with Vigenere Encryption..
+     */
+    public String encodeMessageWithVigenereEncryption(String message, String key) {
+
+        return message == null || message.trim().isEmpty()
+                ? ""
+                : Arrays.stream(message.split(""))
+                .map(letter -> this.isValidCharacter(letter)
+                        ? this.getLetterWithVigenereEncryption(letter, key, true)
+                        : letter)
+                .collect(Collectors.joining());
+    }
+
+    /**
+     * Method to decode Message with Vigenere Encryption.
+     *
+     * @param message This is message to decode with Vigenere Encryption.
+     * @param key     This is a key with string value.
+     * @return String , string already encode with Vigenere Encryption..
+     */
+    public String decodeMessageWithVigenereEncryption(String message, String key) {
+
+        return message == null || message.trim().isEmpty()
+                ? ""
+                : Arrays.stream(message.split(""))
+                .map(letter -> this.isValidCharacter(letter)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3358)